frontend/public/components/factory/table.tsx

```diff
@@ -454,11 +481,18 @@ export const Table = connect<
           sortBy = { index: columnIndex + this._columnShift, direction: currentSortOrder };
         }
       }
-      this.state = { columns, sortBy };
+      this.state = { sortBy };
     }
 
     componentDidMount() {
-      const { columns } = this.state;
+      const componentProps: any = _.pick(this.props, [
```

Comment:
We should avoid `any` type.

Reply:
I didn't write this one and I'm honestly not sure how to better define the type given that it changes so much in action.